### ISSUE TITLE
Remove ingestion-time partitioned tables in BigQuery and all associated logic

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/dbt#2332]
### Describe the feature
dbt's support for "ingestion-time" partitioned tables is a relic of BigQuery's historical lack of support for table partitioning DDL. BigQuery has supported column-level partitioning for some time now, so we should remove the largely-unused ingestion-time partitioning table logic from dbt.

Methods to remove:
 - [impl._materialize_as_table](https://github.com/fishtown-analytics/dbt/blob/dev/octavius-catto/plugins/bigquery/dbt/adapters/bigquery/impl.py#L392)
 - [impl._materialized_as_view](https://github.com/fishtown-analytics/dbt/blob/dev/octavius-catto/plugins/bigquery/dbt/adapters/bigquery/impl.py#L377)
 - [impl.execute_model](https://github.com/fishtown-analytics/dbt/blob/51278c0fbf5ba8e0333befdf3dd9f9cd9949fe54/plugins/bigquery/dbt/adapters/bigquery/impl.py#L479)
 - [connections.create_bigquery_table](https://github.com/fishtown-analytics/dbt/blob/51278c0fbf5ba8e0333befdf3dd9f9cd9949fe54/plugins/bigquery/dbt/adapters/bigquery/connections.py#L252)
 - [connections.create_view](https://github.com/fishtown-analytics/dbt/blob/51278c0fbf5ba8e0333befdf3dd9f9cd9949fe54/plugins/bigquery/dbt/adapters/bigquery/connections.py#L269)
 - [connections.create_table](https://github.com/fishtown-analytics/dbt/blob/51278c0fbf5ba8e0333befdf3dd9f9cd9949fe54/plugins/bigquery/dbt/adapters/bigquery/connections.py#L276)
 - [connections.create_date_partitioned_table](https://github.com/fishtown-analytics/dbt/blob/51278c0fbf5ba8e0333befdf3dd9f9cd9949fe54/plugins/bigquery/dbt/adapters/bigquery/connections.py#L291)
 - [macro: make_date_partitioned_table](https://github.com/fishtown-analytics/dbt/blob/51278c0fbf5ba8e0333befdf3dd9f9cd9949fe54/plugins/bigquery/dbt/include/bigquery/macros/materializations/table.sql#L1-L27) + all associated materialization logic

As a part of this change, we should also remove the BigQuery-specific `timeout_seconds` config, which is only used as a part of creating ingestion-time partitioned tables via the BigQuery HTTP API. We should consider supporting a dbt-wide `timeout_seconds` config in the future (this would apply to any query invoked by dbt), so let's continue to support the config in `profiles.yml`, but render a warning at run-start indicating that the config is not used.

> Warning: The configured target specifies a timeout_seconds config, which is currently unused. Support for this config may be removed in a future release of dbt.

### Additional context
BQ-only... we should be able to delete a whole lot of code!

### Who will this benefit?
dbt maintainers/contributors, users who (rightfully) misunderstand the nature of the `timeout_seconds` config

**Comment by rpedela-recurly** [comment]
I am very supportive of a dbt-wide `timeout_seconds`. For my team's use case, we have an hourly Airflow pipeline and any given BigQuery query should complete in a few minutes or less. If it doesn't something went wrong and we would like it to time out. If it doesn't time out, it can cause our pipeline to be delayed. Additionally, we would like to request that the query is cancelled if there is a timeout, or at least cancellation is attempted, and a log message whether or not the query was cancelled successfully. Airflow does have its own timeout functionality, but it isn't obvious how we can cancel the query.

**Comment by drewbanin** [comment]
Thanks @rpedela-recurly! There are two levels of timeouts we could apply:
- run-level
- node-level

I think a run-level config probably makes the most sense as a command-line argument? I imagine the exact timeout required is going to vary a ton by env / model selections / other flags / etc. So, I can picture:

````
dbt run --run-timeout-seconds 3600 
```

I can also imagine node-level timeouts which are better suited for model-level configs, eg:
```
models:
  timeout_seconds: 3600
```

Here, if any model takes more than 3600 seconds to complete, it will be cancelled by dbt. You could equivalently do something like:
```
snapshots:
  timeout_seconds: 3600
```

so you could have different node-level semantics for different node types.

All of this feels pretty sensible to me -- curious if you buy the proposed implementation

**Comment by rpedela-recurly** [comment]
Yeah I like having a global command-line argument, global config option, and a per-model/snapshot config option.

**Comment by leahwicz** [comment]
Work:
- Ripping out old code
- Should we deprecate or just rip it out - `timeout_seconds `? Would cause an error in the profile so lets give a warning instead (more digging needs to be done here b/c it is used somewhere else)
